contracts/service-registry/src/state.rs

```diff
@@ -145,12 +145,46 @@ pub enum AuthorizationState {
     Jailed,
 }
 
-type ServiceName = str;
+pub struct VerifierPerChainIndexes<'a> {
+    pub verifier_address: MultiIndex<
+        'a,
+        (ServiceName, VerifierAddress),
+        (),
+        (ServiceName, ChainName, VerifierAddress),
+    >,
+}
+
+impl<'a> IndexList<()> for VerifierPerChainIndexes<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<()>> + '_> {
+        let v: Vec<&dyn Index<()>> = vec![&self.verifier_address];
+        Box::new(v.into_iter())
+    }
+}
+
+pub const VERIFIERS_PER_CHAIN_INDEXED_MAP: IndexedMap<
```

Comment:
```suggestion
pub const VERIFIERS_PER_CHAIN: IndexedMap<
```

I generally don't like including the type in the name. Anyone can just go to the definition to see the type, or just use their IDE, so it feels unnecessarily verbose.